File: Easy/Fizz_Buzz/Fizz_Buzz.java
import java.util.*;

class Solution {
	
	//Time complexity O(N)
    public List<String> fizzBuzz(int n) {
		
		//Initialize a list. We need java.util package to use List.
        List<String> arrayList = new ArrayList<String>();//List is an interface which means it cannot be instantiated (no new List() is possible).
        for ( int i = 1; i <= n; i++ ){
			// if (i % 3 == 0 && i % 5 == 0) {
			// 	arrayList.add("FizzBuzz");
			// }
			// else if ( i % 3 == 0 ) {
			// 	arrayList.add("Fizz");
			// }
			// else if ( i % 5 == 0) {
			// 	arrayList.add("Buzz");
			// }
			// else {
			// 	arrayList.add(""+i);
			// }  
            String s = "";
            if (i % 3== 0){
                s = s+"Fizz";
            }
            if (i % 5 == 0){
                s = s+"Buzz";
            }
            if (s.equals("")){
                arrayList.add(""+i);
            }
            else {
                arrayList.add(""+s);
            }
        }
        return arrayList;
    }
}

File: Fizz Buzz/Fizz_Buzz.kt
class Solution {
    fun fizzBuzz(n: Int): List<String> {
        var list: MutableList<String> = mutableListOf<String>()
        for ( x in 1 until n+1 ){
            var s: String = ""
            if ( x % 3 == 0 ){
                s += "Fizz"
            }
            if ( x % 5 == 0 ){
                s += "Buzz"
            }
            if ( x % 3 != 0 && x % 5 != 0 ){
                s = x.toString()
            }
            list.add(s)
        }
        return list
    }
}